### PR TITLE
[16.0][FIX] account_payment_order: set invoice name in communication if payment order is inbound

### PR DESCRIPTION
**Pull request by ghost**
Sets invoice name in communication (`account.payment.line`) if payment order is inbound.

### CODE REVIEW

File: account_payment_order/models/account_move.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ def _get_payment_order_communication_direct(self):
                 communication = self.ref or self.payment_reference
             else:
                 communication = self.payment_reference or self.name
+        default_payment_type = self._context.get("default_payment_type", False)
+        if default_payment_type == "inbound":
+            communication = self.name
```
Comment (aritzolea) [inline account_payment_order/models/account_move.py]: Should the name also de assigned if the account.move is not an invoice?

Comment (ghost) [reply account_payment_order/models/account_move.py]: Changed so that this only applies when it's an invoice

**Review by aritzolea — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Remove draft state when ready to be reviewed

**Review by SoniaViciana — APPROVED** [review]
Functional review done. With this change, when you create a payment order (charge), the communication is the invoice number instead of the sale order number. I think it makes more sense.

Thanks!

**Comment by ghost** [comment]
Hello @OCA/banking-maintainers, 

Could you review this? 

Thank you!

**Review by alexey-pelykh — COMMENTED** [review]
could you please also add tests without default_payment_type in context to ensure nothing breaks for those cases?

File: account_payment_order/models/account_move.py [hunk 2]
```diff
@@ -57,6 +57,16 @@ def _get_payment_order_communication_direct(self):
                 communication = self.ref or self.payment_reference
             else:
                 communication = self.payment_reference or self.name
+            default_payment_type = self._context.get("default_payment_type", False)
```
Comment (etobella) [inline account_payment_order/models/account_move.py:58]: Why are you using the default_payment_type?

If you want to use name, just leave payment_reference and ref to false :thinking:

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
This is not correct.

The payment reference field is clear: it's for setting the reference of the payment. Setting it, it should have preference over the invoice number. And anyway, by default, the payment reference is set with the invoice number.

Which is the problem this is trying to fix?

**Comment by SoniaViciana** [comment]
When this PR was proposed the "payment reference" was completed by default with the sales order number instead of invoice number.

This should have been solved in parallel since it does not happen today:

image

We can close this issue.

**Comment by pedrobaeza** [comment]
For other times, it's good to add proper explanations since the beginning and in the commit message, so when recalling this, the reasons are clear and can be contrasted.

**Pull request closed without merging**